lib/src/local_working_copy.rs

```diff
@@ -1513,14 +1514,36 @@ impl FileSnapshotter<'_> {
             let id = self.write_file_to_store(repo_path, disk_path).await?;
             // On Windows, we preserve the executable bit from the current tree.
             let executable = executable.unwrap_or_else(|| {
-                if let Some(TreeValue::File { id: _, executable }) = current_tree_value {
+                if let Some(TreeValue::File {
+                    id: _,
+                    executable,
+                    copy_id: _,
+                }) = current_tree_value
+                {
                     *executable
                 } else {
                     false
                 }
             });
-            Ok(Merge::normal(TreeValue::File { id, executable }))
-        } else if let Some(old_file_ids) = current_tree_values.to_file_merge() {
+            // Preserve the copy id from the current tree
+            let copy_id = {
+                if let Some(TreeValue::File {
+                    id: _,
+                    executable: _,
+                    copy_id,
+                }) = current_tree_value
+                {
+                    copy_id.clone()
+                } else {
+                    CopyId::placeholder()
```

Comment:
I'm not sure if this `placeholder` is a temporary stub or a marker to generate new `salt`-ed `CopyHistory`. Maybe leave a TODO comment?

Reply:
True, this is one of few places where we will create a new copy id.